app/controllers/webauthn_credentials_controller.rb

```diff
@@ -0,0 +1,48 @@
+class WebauthnCredentialsController < ApplicationController
+  before_action :redirect_to_signin, unless: :signed_in?
+
+  def create
+    @create_options = current_user.webauthn_options_for_create
+
+    session[:webauthn_registration] = { challenge: @create_options.challenge }
+
+    render json: @create_options
+  end
+
+  def callback
+    credential = WebAuthn::Credential.from_create(params.require(:credentials))
+    credential.verify(session.dig(:webauthn_registration, "challenge"))
+
+    webauthn_credential = current_user.webauthn_credentials.build(
+      webauthn_credential_params.merge(
+        external_id: credential.id,
+        public_key: credential.public_key,
+        sign_count: credential.sign_count
+      )
+    )
+
+    if webauthn_credential.save
+      render json: { location: edit_settings_path }
+    else
+      render(
+        json: { message: webauthn_credential.errors.full_messages.to_sentence },
+        status: :unprocessable_entity
+      )
+    end
+  rescue WebAuthn::Error => e
+    render json: { message: e.message }, status: :unprocessable_entity
```

Comment:
I guess it could be translated but I didn't find how to trigger it from a regular user experience (e.g. not sending random data through curl)